File: cf/py/1748B/main.py
# This Code TLE!
# See go code: github.com/huxulm/apsr/cf/go/1748B/main.go
IS = lambda: input()
II = lambda: int(IS())
LI = lambda: map(int, IS().split())
def solve():
  t = II()
  for _ in range(t):
    ans = 0
    n, s = II(), IS()
    for i in range(n):
      cnt = [0] * 10
      k = mxC = 0
      for j in range(i, n):
        v = ord(s[j]) - ord('0')
        if cnt[v] == 10:
          break
        k += cnt[v] == 0
        cnt[v] += 1
        mxC = max(mxC, cnt[v])
        ans += mxC <= k
    print(ans)

if __name__ == '__main__':
  solve()

File: cf/go/1748B/main.go
// https://www.luogu.com.cn/problem/CF1748B
package main

import (
	"fmt"
)

func main() {
	var (
		t, n int
		s    string
	)

	fmt.Scan(&t)
	for i := 0; i < t; i++ {
		fmt.Scan(&n, &s)
		var ans int
		for i := range s {
			var k, mxC int
			var cnt [10]int
			for _, v := range s[i:] {
				v -= '0'
				if cnt[v] == 10 {
					break
				}
				if cnt[v] == 0 {
					k++
				}
				cnt[v]++
				mxC = max(mxC, cnt[v])
				if mxC <= k {
					ans++
				}
			}
		}
		fmt.Println(ans)
	}
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
